libs/Microsoft.MixedReality.WebRTC/PeerConnection.cs

```diff
@@ -1214,6 +1214,117 @@ public void SetRemoteDescription(string type, string sdp)
+            public ulong BytesSent;
+        }
+
+        /// <summary>
+        /// RTCMediaStreamTrack (audio receiver subset) + RTCInboundRTPStreamStats.
+        /// </summary>
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+        public struct AudioReceiverStats
+        {
+            public long TimestampUs;
+            public string TrackIdentifier;
+            public double AudioLevel;
+            public double TotalAudioEnergy;
+            public double TotalSamplesReceived;
+            public double TotalSamplesDuration;
+
+            public long RtpStatsTimestampUs;
+            public uint PacketsReceived;
+            public ulong BytesReceived;
+        }
+
+        /// <summary>
+        /// RTCMediaStreamTrack (video sender subset) + RTCOutboundRTPStreamStats.
+        /// </summary>
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+        public struct VideoSenderStats
+        {
+            public long TimestampUs;
+            public string TrackIdentifier;
+            public uint FramesSent;
+            public uint HugeFramesSent;
+
+            public long RtpStatsTimestampUs;
+            public uint PacketsSent;
+            public ulong BytesSent;
+            public uint FramesEncoded;
+        }
+
+        /// <summary>
+        /// RTCMediaStreamTrack (video receiver subset) + RTCInboundRTPStreamStats.
+        /// </summary>
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+        public struct VideoReceiverStats
+        {
+            public long TimestampUs;
+            public string TrackIdentifier;
+            public uint FramesReceived;
+            public uint FramesDecoded;
+            public uint FramesDropped;
+
+            public long RtpStatsTimestampUs;
+            public uint PacketsReceived;
+            public ulong BytesReceived;
+        }
+
+        /// <summary>
+        /// RTCTransportStats
+        /// </summary>
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+        public struct TransportStats
+        {
+            public long TimestampUs;
+            public ulong BytesSent;
+            public ulong BytesReceived;
+        }
+
+        public interface StatsCollector
+        {
+            void Handle(in DataChannelStats stats);
+            void Handle(in AudioSenderStats stats);
+            void Handle(in AudioReceiverStats stats);
+            void Handle(in VideoSenderStats stats);
+            void Handle(in VideoReceiverStats stats);
+            void Handle(in TransportStats stats);
+        }
+
+        public void GetSimpleStats(StatsCollector collector)
+        {
+            var collectorHandle = GCHandle.Alloc(collector, GCHandleType.Normal);
+            PeerConnectionInterop.PeerConnection_GetSimpleStats(_nativePeerhandle,
+                PeerConnectionInterop.SimpleStatsCallback, GCHandle.ToIntPtr(collectorHandle));
+        }
```

Comment:
Is that staying as it or changing to something more user friendly like `PeerConnection.GetStats(out TransportStats)` or `LocalVideoTrack.GetStats(out VideoSenderStats)`?

Reply:
Changed to an IEnumerable based interface.